EKrut-Server/src/ekrut/server/managers/ServerShipmentManager.java

```diff
@@ -1,53 +1,148 @@
+import ekrut.server.db.DBController;
+import ekrut.server.db.OrderDAO;
 import ekrut.server.intefaces.ShipmentManagerUtils;
 
+/**
+ * The {@code ServerShipmentManager} class is responsible for managing the
+ * server shipping requests. This includes fetching shipment requests for
+ * approval, confirming the shipment, confirming the delivery, and marking the
+ * order as done.
+ * 
+ * @author Nir Betesh
+ */
 public class ServerShipmentManager {
-	
+
+	OrderDAO orderDAO;
+
+	/**
+	 * 
+	 * Constructs a new {@code ServerShipmentManager} object.
+	 * 
+	 * @param con the database controller to use for accessing the database
+	 */
+	public ServerShipmentManager(DBController con) {
+		orderDAO = new OrderDAO(con);
+	}
+
+	// Q.Nir , TBD - Need to change after Yovel implementation?
+	// PLEASE, HOW CAN I DO IT?
+	// C.Nir - Message need to be sent after confirmation.
+
+	/**
+	 * Fetches a list of shipment requests for approval.
+	 * 
+	 * @param shipmentRequest the {@code ShipmentRequest} object containing the request details
+	 * @return a {@code ShipmentResponse} object with the result of the operation
+	 * @throws IllegalArgumentException if {@code shipmentRequest} is {@code null}
+	 */
+	public ShipmentResponse fetchShipmentRequests(ShipmentRequest shipmentRequest) {
+		if (shipmentRequest == null)
+			throw new IllegalArgumentException("null shipmentRequest was provided.");
+
+		ArrayList<Order> ShippingForAppoval = orderDAO.fetchOrderShipmentList(shipmentRequest.getArea());
+		if (ShippingForAppoval == null)
+			return new ShipmentResponse(ResultType.NOT_FOUND);
+
+		return new ShipmentResponse(ResultType.OK);
+	}
+
 	/**
-	 * Create shipping request for 
-	 * @param shipmentRequest
-	 * @return return if shipping request was successfully passed
+	 * Confirms the shipment of an order by EKurt worker.
+	 * 
+	 * @param shipmentRequest the {@code ShipmentRequest} object containing the request details
+	 * @return a {@code ShipmentResponse} object with the result of the operation
+	 * @throws IllegalArgumentException if {@code shipmentRequest} is {@code null}
 	 */
-	public ShipmentResponse createShippingRequest(ShipmentRequest shipmentRequest) {
+	public ShipmentResponse confirmShipment(ShipmentRequest shipmentRequest) {
 		if (shipmentRequest == null)
-			throw new IllegalArgumentException("Null order was provided");
-		
+			throw new IllegalArgumentException("Null order was provided.");
+
 		int orderId = shipmentRequest.getOrderId();
 		LocalDateTime date = shipmentRequest.getDate();
 		String clientAddress = shipmentRequest.getClientAddress();
-		
-		// estimate delivery time.
+		Order order = orderDAO.fetchOrderById(orderId);
+
+		// In case the order null an exception will be thrown
+		if (order == null)
+			return new ShipmentResponse(ResultType.NOT_FOUND);
+
+		if (order.getType() != OrderType.SHIPMENT)
+			return new ShipmentResponse(ResultType.INVALID_INPUT);
+
+		if (order.getStatus() != OrderStatus.AWAITING_DELIVERY)
```

Comment:
If the order was yet to be approved by the worker, it will be in the `SUBMITTED` status.

Reply:
Fixed

Reply:
You changed the status in the wrong spot.

Reply:
-_-